packages/ui/src/views/unified-pipeline-studio/components/visual-yaml-toggle.tsx

```diff
@@ -12,36 +10,19 @@ interface VisualYamlToggleProps {
 }
 
 export const VisualYamlToggle = (props: VisualYamlToggleProps): JSX.Element => {
-  const { view, setView, isYamlValid, className } = props
-  const { isLightTheme } = useTheme()
+  const { view, setView, isYamlValid } = props
+
+  const onChange = (value: VisualYamlValue) => {
+    setView(value)
+  }
+
   return (
-    <ToggleGroup.Root
-      className={cn(
-        'h-8 rounded-md border px-1 bg-cn-background-3 inline-flex',
-        isLightTheme ? 'border-cn-borders-5' : 'border-cn-borders-2',
-        className
-      )}
-      onValueChange={value => {
-        if (value) {
-          setView(value as VisualYamlValue)
-        }
-      }}
-      value={view}
-      type="single"
-      unselectable="on"
-    >
-      <ToggleGroup.Item
-        disabled={!isYamlValid}
-        value="visual"
-        className="rounded text-2 disabled:opacity-100"
-        size="xs"
-      >
-        {!isYamlValid && <Icon name="fail-legacy" className="mr-1 text-cn-foreground-danger" />}
+    <ToggleGroup.Root variant="ghost-secondary" onValueChange={onChange} value={view} size="xs" unselectable>
+      <ToggleGroup.Item disabled={!isYamlValid} value="visual">
+        {!isYamlValid && <Icon name="fail-legacy" className="text-cn-foreground-danger" />}
```

Comment:
not a blocker for this PR, but @praneshg239 can we consider adding a `theme` prop on the `Icon` component? I think showing icons in green/red/blue will be a common usecase.